src/app/shared/components/template/pipes/qr-code.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from "@angular/core";
+import QRCode from "qrcode";
+
+@Pipe({
+  name: "qrCode",
+})
+export class QRCodePipe implements PipeTransform {
+  async transform(value: any, args?: any[]): Promise<any> {
+    if (value && typeof value === "string" && value.length >= 0) {
```

Comment:
nit(non-blocking) - if the value exists and is a string then you shouldn't need to also check its length. A zero-length string `""` is falsy so will fail the first clause, and so what you are left with will have length >0 regardless